src/psyclone/parse/kernel.py

```diff
@@ -468,7 +472,22 @@ def mesh(self):
         '''
         return self._mesh
 
+    @property
+    def type(self):
```

Comment:
I would rather go for  `arg_type` or `argument_type` as it maps nicely into the names used in various `psyGen` functions. This, of course, means that it would be god to change `type` to something else not only in the `Descriptor` and the children classes but also in `Argument` and the related children classes. It is a lot of code changes as `descriptor.type` and `argument.type` are heavily used!

Reply:
Changed to `argument_type` as this name is used a lot in `psyGen` for applications of this property.